### DISCUSSION TITLE
Does the maximized window have a callback method ？

### DISCUSSION THREAD

**Discussion by YTLoudYN** [beeware/toga#1967] [Q&A]
https://github.com/beeware/toga/assets/131869902/b6d49a5f-28be-4365-8e9c-abb87af18ed2

**Comment by freakboy3742** [comment]
It's very difficult to provide assistance when all we have to go on is a video. It might be obvious to you what is going on, but we don't know anything about what your app is *meant* to do.

If you're referring to the fact that the WebView isn't resizing when you maximise the window, it's possible you may have found a bug - however, it's impossible to say when all you've provided is a video. It's possible that the issue is in your Toga code, or in the code of your web content.

**Reply by freakboy3742** [reply]
That is definitely some code. However, it would appear to be incomplete - as written, there's nothing in the main box (and no startup method either). You've redacted so much that it's no longer a working example.

What is it meant to do? What behavior do you expect?

**Reply by YTLoudYN** [reply]
I cut out part of the code

**Reply by freakboy3742** [reply]
That much is obvious. However, as a result, the code *doesn't do anything*. It's no longer an adequate explanation of the problem you're having.

I can't provide a solution to your problem until I *understand* your problem. And that means you either have to provide a text description, or *working* code.

**Reply by freakboy3742** [reply]
In order to solve your problem, I need to be able to understand it. Dumping the complete source code for a large complex project isn't going to help me - especially when that code has been redacted, and won't run as provided.

Step one is for you to either:
1) Generate a *minimal* reduced example that will run *as is*, or
2) Write an explanation of what it is you're trying to do, and what isn't working.

Until you do one of these two things, I am *unable to help you*.

As an aside, the process of generating (1) will, quite often, reveal the source of the problem, because you'll remove something that is causing a complication.

As a second aside - Github has a Preview button. You have now posted multiple examples of code, and every time you've made the same markup error, which makes it very difficult to read your code. Please ask that you preview your posts before posting - or, if something doesn't render right when posting, use the *edit* button to correct the markup.

**Comment by YTLoudYN** [comment]
It can now run
```
import toga
from toga.style import Pack
from toga.constants import *
from http.server import HTTPServer, SimpleHTTPRequestHandler
from threading import Event, Thread

class HTTPHandler(SimpleHTTPRequestHandler):
    def translate_path(self, path):
        return str(self.server.base_path / path[1:])

class LocalHTTPServer(HTTPServer):
    def __init__(self, base_path, RequestHandlerClass=HTTPHandler):
        self.base_path = base_path
        # Use port 0 to let the server select an available port.
        super().__init__(("127.0.0.1", 0), RequestHandlerClass)

class AliShare(toga.App):

    def startup(self):
        self.main_box = toga.Box(style=Pack(direction=COLUMN,flex=1))

        self.optioncontainer = toga.OptionContainer(style=Pack(flex=1))
        self.optioncontainer.add('网页版', self.get_op_5())
        self.main_box.add(self.optioncontainer)

        self.main_window = toga.MainWindow(title=self.formal_name,size=(510,600),position=(100,0))
        self.main_window.content = self.main_box
        self.main_window.show()

    def get_op_5(self):
        self.server_exists = Event()

        self.web_view = toga.WebView(style=Pack(flex=1))

        self.server_thread = Thread(target=self.web_server)
        self.server_thread.start()

        self.on_exit = self.cleanup

        self.server_exists.wait()
        self.web_view.url = 'https://www.aliyundrive.com/drive'
        return self.web_view

    def web_server(self):
        print("Starting server...")
        self._httpd = LocalHTTPServer(self.paths.app / "resources" / "webapp")
        self.server_exists.set()
        self._httpd.serve_forever()

    def cleanup(self, app, **kwargs):
        print("Shutting down...")
        self._httpd.shutdown()
        return True


def main():
    return AliShare('Ali Share','id')

if __name__ == '__main__':
    main().main_loop()
```

**Reply by freakboy3742** [reply]
I'll note that a lot of that code isn't needed - there's no need to start a web server to generate this problem, for example - but I can now reproduce the problem you're describing.

It appears to be a bug on Windows; when using the Maximise button, the OptionContainer isn't being instructed to re-compute its layout. I've opened #1968 to track this problem.
